### DISCUSSION TITLE
Variables not working for "where"

### DISCUSSION THREAD

**Discussion by kavatari** [ChilliCream/hotchocolate#7877] [Q&A] [answered]
I have the following query, which works fine in nitro and returns the expected result:  
```gql
query DocumentQuery{
  documents(order: [ {
      partnerName: ASC
  }], 
  skip: 0, 
  take: 10, 
  where:  {
     or: [ 
      {
        description:  {
           contains: "test"
        },
     }
     ]
  }){
    totalCount,
    items {
      id,
      number,
      description,
      partnerName,
      createdAt
    }
  },
}
```
![image](https://github.com/user-attachments/assets/022f9cdc-060c-449c-9c5b-5601f8406bda)

But, when using this query in Apollo Vue Client (https://apollo.vuejs.org), the Hot Chocolate server returns "HC0047".

Non working example: 
```typescript
const { loading, onError, onResult, load } = useLazyQuery<DocumentQuery>(
  gql`
     query DocumentQuery($order: [DocumentQueryDtoSortInput!], $skip: Int!, $take: Int!, $where: DocumentQueryDtoFilterInput) {
      documents(order: $order, skip: $skip, take: $take, where: $where) {
          items {
            id,
            number,
            description,
            partnerName,
            createdAt
          },
          totalCount
      }
     }
    `, () => ({
    order: [getOrderBy.value],
    skip: (pagination.value.page - 1) * pagination.value.rowsPerPage,
    take: pagination.value.rowsPerPage,
    where:  filterOptions.value
  })
);
```

On the other hand, when removing the "where" from the query above, like this: it works! 
working example: 
```typescript
const { loading, onError, onResult, load } = useLazyQuery<DocumentQuery>(
  gql`
     query DocumentQuery($order: [DocumentQueryDtoSortInput!], $skip: Int!, $take: Int!) {
      documents(order: $order, skip: $skip, take: $take) {
          items {
            id,
            number,
            description,
            partnerName,
            createdAt
          },
          totalCount
      }
     }
    `, () => ({
    order: [getOrderBy.value],
    skip: (pagination.value.page - 1) * pagination.value.rowsPerPage,
    take: pagination.value.rowsPerPage
  })
);
```
![image](https://github.com/user-attachments/assets/8f24d5d7-49b3-463b-8053-3648649f8921)

Also, looking at the above screenshots while debugging the hc GraphQL context, you can see that the variables are not available on the "context" level, but only on the request level in the non working example, but in  the working example they are on the "context" level. 

What am I doing wrong here?

**Comment by kavatari** [comment]
I am having the same error "HC0047" using Nitro and variables this way: 
```gql 
query DocumentQuery($where: DocumentQueryDtoFilterInput){
  documents(order: [ {
      partnerName: ASC
  }], 
  skip: 0, 
  take: 10, 
  where:  $where){
    totalCount,
    items {
      id,
      number,
      description,
      partnerName,
      createdAt
    }
  },
}
```
// Vars
```json
{
  "where" : {
     "or": [ 
      {
        "description":  {
           "contains": "test"
        }
     }
     ]
  }
}
```

**Comment by RenatV** [comment]
Maybe it helps:
```
builder.Services
    .AddGraphQLServer()
    .ModifyCostOptions(o => o.EnforceCostLimits = false)
```
But read https://chillicream.com/docs/hotchocolate/v14/security/cost-analysis#applying-a-cost-weight  before disable cost limits.

**Reply by kavatari** [reply]
@RenatV thanks for the your reply. Just tested the "EnforceCostLimits = false" option, which resolves the HC0047 error. 
But the variables are still not validated properly, because they are not properly processed like I showed in the screenshots before.

**Comment by kavatari** [comment, marked as answer]
Resolved the issue using "FilterInputType", see https://chillicream.com/docs/hotchocolate/v14/fetching-data/filtering
It looks like I need to narrow down the props that can be used for filtering and ordering.
